march_rqt_gait_generator/src/march_rqt_gait_generator/JointSettingPlot.py

```diff
@@ -0,0 +1,191 @@
+
+        self.dragPoint = None
+        self.dragIndex = -1
+        self.dragOffset = 0
+        self.plot_item = None
+        self.plot_interpolation = None
+
+        self.velocity_markers = []
+
+        # Store the velocities so a plot can be converted to a setpoint.
+        self.velocities = []
+
+        self.lower_limit = math.degrees(joint.limits.lower)
+        self.upper_limit = math.degrees(joint.limits.upper)
+        self.duration = duration
+
+        self.createPlots(joint)
+
+        self.setTitle(joint.name)
+
+        self.setYRange(self.lower_limit-0.1, self.upper_limit+0.1, padding=0)
+        limit_pen = pg.mkPen(color='r', style=QtCore.Qt.DotLine)
+        self.addItem(pg.InfiniteLine(self.lower_limit, angle=0, pen=limit_pen))
+        self.addItem(pg.InfiniteLine(self.upper_limit, angle=0, pen=limit_pen))
+        self.setXRange(-0.1, self.duration + 0.1, padding=0)
+        self.setMouseEnabled(False, False)
+        self.setMenuEnabled(False)
+        self.hideButtons()
+
+        self.updateSetpoints(joint)
+
+        time_pen = pg.mkPen(color='y', style=QtCore.Qt.DotLine)
+        self.time_line = self.addLine(0, pen=time_pen, bounds=(0, self.duration))
+
+    def createVelocityMarkers(self, setpoints):
+        # Remove old sliders
+        while self.velocity_markers:
+            self.removeItem(self.velocity_markers.pop())
+            self.velocities.pop()
+
+        marker_length = self.duration/10.0
+
+        for setpoint in setpoints:
+            velocity_pen = pg.mkPen(color='g', size=3)
+
+            # Calculate start and endpoint of velocity marker
+            dx = 0.5*marker_length*math.cos(math.atan(setpoint.velocity))
+            x_start = setpoint.time - dx
+            x_end = setpoint.time + dx
+
+            dy = math.degrees(0.5*marker_length*math.sin(math.atan(setpoint.velocity)))
+            y_start = math.degrees(setpoint.position) - dy
+            y_end = math.degrees(setpoint.position) + dy
+
+            self.velocity_markers.append(self.plot([x_start, x_end], [y_start, y_end], pen=velocity_pen))
+            self.velocities.append(setpoint.velocity)
+
+    def updateTimeSlider(self, time):
+        self.time_line.setValue(time)
+
+    def createPlots(self, joint):
+        self.plot_item = self.plot(pen=None, symbolBrush=(255, 0, 0), symbolPen='w')
+        self.showGrid(True, True, 1)
+        self.plot_interpolation = self.plot()
+
+    def updateSetpoints(self, joint):
+        time, position, velocity = joint.get_setpoints_unzipped()
+
+        for i in range(0, len(position)):
+            position[i] = math.degrees(position[i])
+
+        self.createVelocityMarkers(joint.setpoints)
+
+        self.plot_item.setData(time, position)
+
+        [indices, values] = joint.interpolate_setpoints()
+        for i in range(0, len(values)):
+            values[i] = math.degrees(values[i])
+
+        self.plot_interpolation.setData(indices, values)
+
+    def mouseClickEvent(self, ev):
```

Comment:
why do you use 'ev' instead of 'event' I assume?

Reply:
Copy paste error, fixed now.